baseapp/files/rest_framework/utils.py

```
@@ -0,0 +1,31 @@
+def enforce_can_attach_to_parent(user, parent_document_pk) -> None:
+    """
+    Authorize attaching a file to the object behind ``parent_document_pk`` (a
+    ``DocumentId`` pk). Mirrors the ``FileAttachToTarget`` GraphQL mutation so
+    the REST upload/set-parent paths cannot bypass target-level authorization:
+    the target must opt into files (``FileableModel``) and the user must hold
+    the ``add_<file_model>`` permission on it.
+    """
+    try:
+        document = DocumentId.objects.get(pk=parent_document_pk)
+    except DocumentId.DoesNotExist:
+        raise ValidationError("Invalid parent: target not found.")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)